test/run-spec-wasm2c.py

```diff
@@ -122,10 +134,10 @@ def __init__(self, spec_json, prefix, out_file, out_dir):
         self.module_name_to_idx = {}
         self.module_prefix_map = {}
         self.unmangled_names = {}
-
-    def Write(self):
         self._MaybeWriteDummyModule()
```

Comment:
Why move this line from `Write` to `__init__`.. doesn't it make sense to do all the writing in `Write`?

Reply:
Unfortunately not in this case -- `_MaybeWriteDummyModule` is doing a job here that's normally done by `wast2json` (writing out a wasm file). We need this to happen before `_CacheModulePrefixes` (so even the dummy module prefix is mapped) and that needs to happen before calling `GetModulePrefixUnmangled`, which is needed to run `wasm2c`.

And (and here's the change actually relevant to this PR) we need `wasm2c` to run before `cwriter.Write()`, because `_WriteModuleInstanceImports` wants to read the `.h` output of `wasm2c` to know which module types are needed in the arguments to the instance init function. These are the "imported-from" modules.